internal/ingress/controller/controller.go

```diff
@@ -1033,6 +1017,11 @@ func (n *NGINXController) createServers(data []*ingress.Ingress,
 				SSLPassthrough: anns.SSLPassthrough,
 				SSLCiphers:     anns.SSLCiphers,
 			}
+
+			// setup default SSL certificate
+			if host == defServerName {
+				servers[host].SSLCert = n.getDefaultSSLCertificate()
+			}
```

Comment:
why are you not doing this in https://github.com/kubernetes/ingress-nginx/pull/4356/files#diff-86ac9ff9d75a0c5005c116e766a6127dL951?

Reply:
good point.